lib/azure/armrest/environment.rb

```diff
@@ -0,0 +1,124 @@
+module Azure
+  module Armrest
+    class Environment
+      # A list of valid keys that can be passed to the constructor
+      VALID_KEYS = [ 
+        :name,
+        :active_directory_authority,
+        :active_directory_resource_id,
+        :gallery_url,
+        :graph_url,
+        :graph_api_version,
+        :key_vault_dns_suffix,
+        :key_vault_service_resource_id,
+        :publish_settings_file_url,
+        :resource_manager_url,
+        :service_management_url,
+        :sql_database_dns_suffix,
+        :storage_suffix,
+        :traffic_manager_dns_suffix
+      ]
+
+      # The Environment name
+      attr_reader :name
```

Comment:
How about `VALID_KEYS.each { |k| attr_reader k }` to avoid repeatedly typing these keys?